tools/slang-unit-test/unit-test-coverage-metal-runtime.cpp

```
@@ -0,0 +1,403 @@
+    targetDesc.format = SLANG_METAL;
+    targetDesc.profile = globalSession->findProfile("metal");
+
+    // Enable all three coverage modes so line, function, and branch-arm
+    // counters are all validated against the same GPU dispatch. Pin the
+    // counter width to 4 bytes: MSL only provides 32-bit
+    // `atomic_fetch_add_explicit`, so emitting the default 64-bit counters
+    // would fail in the Metal compiler with "no matching function for call
+    // to 'atomic_fetch_add_explicit'".
+    slang::CompilerOptionEntry coverageOptions[4] = {};
+    uint32_t coverageOptionCount = 0;
+    auto addIntOption = [&](slang::CompilerOptionName name, int value)
```

Cppcheck (2.21.0)

[style] The function 'writeRef' is never used.

(unusedFunction)